### PR TITLE
Modify run.py to use torch.profiler

### PR DESCRIPTION
**Pull request by aaronenyeshi**
Deprecate the use of torch.autograd.profiler, and use
torch.profiler. Clean up a few linting errors. Print
Tensorboard traces to a default folder, and add an
option for the user to configure the trace folder.

### CODE REVIEW

**Comment by aaronenyeshi** [comment]
The CI seems to be failing on an un-related issue. Any idea how to get past this error?
```
+ git lfs fetch
fetch: Fetching reference refs/heads/aaron-cleanup-run
[e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855] Object does not exist on the server: [404] Object does not exist on the server
error: failed to fetch some objects from 'https://github.com/pytorch/benchmark.git/info/lfs'

Exited with code exit status 2
CircleCI received exit code 2
```

File: run.py [hunk 1]
```diff
@@ -50,56 +51,68 @@ def run_one_step_with_cudastreams(func, streamcount):
...
+    activity_groups = []
+    if args.device == "cuda":
+        activity_groups.append(profiler.ProfilerActivity.CUDA)
+    activity_groups.append(profiler.ProfilerActivity.CPU)
+
+    with profiler.profile(
+        schedule=profiler.schedule(wait=0, warmup=0, active=1),
+        activities=activity_groups,
+        record_shapes=True,
+        on_trace_ready=profiler.tensorboard_trace_handler('./logs')
```
Comment (wconstab) [inline run.py:96]: shouldn't you take args.profile-folder here instead of './logs'?

**Review by wconstab — APPROVED** [review]
lgtm except the one issue I pointed out, stamping now to unblock but please fix that

File: run.py (hunk 1 above)
Comment (aaronenyeshi) [reply run.py:96]: Good catch, thanks that was remains from my testing. fixing it!

**Comment by aaronenyeshi** [comment]
> lgtm except the one issue I pointed out, stamping now to unblock but please fix that

Thanks, fixed the issue. Looks like this needs to be imported internally for merge. will do that once the import option comes back up.

**Comment by aaronenyeshi** [comment]
Rebased due to CI errors yesterday. Also moved the warmup loop inside of prof block, as intended by profiler APIs.

File: run.py [hunk 2]
```diff
@@ -50,56 +51,67 @@ def run_one_step_with_cudastreams(func, streamcount):
...
     parser = argparse.ArgumentParser(__doc__)
     parser.add_argument("model", help="Full or partial name of a model to run.  If partial, picks the first match.")
-    parser.add_argument("-d", "--device", choices=["cpu",  "cuda"], default="cpu", help="Which device to use.")
-    parser.add_argument("-m", "--mode", choices=["eager",  "jit"], default="eager", help="Which mode to run.")
-    parser.add_argument("-t", "--test", choices=["eval",  "train"], default="eval", help="Which test to run.")
+    parser.add_argument("-d", "--device", choices=["cpu", "cuda"], default="cpu", help="Which device to use.")
+    parser.add_argument("-m", "--mode", choices=["eager", "jit"], default="eager", help="Which mode to run.")
+    parser.add_argument("-t", "--test", choices=["eval", "train"], default="eval", help="Which test to run.")
     parser.add_argument("--profile", action="store_true", help="Run the profiler around the function")
+    parser.add_argument("--profile-folder", default="./logs", help="Save profiling model traces to this directory.")
```
Comment (xuzhao9) [inline run.py:110]: Please add "logs/" to `.gitignore` to prevent future possible commits of the directory.

File: run.py [hunk 3]
```diff
@@ -50,56 +51,67 @@ def run_one_step_with_cudastreams(func, streamcount):
...
+
+    with profiler.profile(
+        schedule=profiler.schedule(wait=0, warmup=nwarmup, active=1),
+        activities=activity_groups,
+        record_shapes=True,
+        on_trace_ready=profiler.tensorboard_trace_handler(args.profile_folder)
+    ) as prof:
+        for _i in range(nwarmup + 1):
+            func()
+            prof.step()
```
Comment (xuzhao9) [inline run.py:97]: Just curious that you are now running `prof.step()` with warmup loops, is there a reason for that?

**Review by xuzhao9 — APPROVED** [review]
LGTM!

File: run.py (hunk 2 above)
Comment (aaronenyeshi) [reply run.py:110]: Will do!

File: run.py (hunk 3 above)
Comment (aaronenyeshi) [reply run.py:97]: I think both ways work the same, I need to look into if profiler does anything special for warmup steps. I think this is the expected way to use `torch.profiler` from the docs. This change reflects the docs better, but is not necessary. Do you have a preference? I don't mind changing it back.

Comment (xuzhao9) [reply run.py:97]: Either way looks okay to me too :)

**Comment by aaronenyeshi** [comment]
@erichan1 and @xuzhao9 , this PR is merged internally, but not reflected on the `main` branch in Github. Any idea what went wrong? (In kineto, the main branch is updated with the commit before closing the PR).

**Pull request closed without merging**